src/mono/mono/metadata/assembly-internals.h

```diff
@@ -13,8 +13,10 @@
 
 #ifndef ENABLE_NETCORE
 #define MONO_ASSEMBLY_CORLIB_NAME "mscorlib"
+#define MONO_ASSEMBLY_CORLIB_RESOURCE_NAME "mscorlib.resources"
 #else
 #define MONO_ASSEMBLY_CORLIB_NAME "System.Private.CoreLib"
+#define MONO_ASSEMBLY_CORLIB_RESOURCE_NAME "System.Private.CoreLib.resources"
 #endif
```

Comment:
```suggestion
#else
#define MONO_ASSEMBLY_CORLIB_NAME "System.Private.CoreLib"
#endif
#define MONO_ASSEMBLY_CORLIB_RESOURCE_NAME (MONO_ASSEMBLY_CORLIB_NAME ".resources")
```

not sure about this.  string literal concatenation might be a bit too esoteric.

Reply:
Yeah, I had that at first and decided it wasn't worth it. I have no strong opinion though, so happy to switch back.

Reply:
Also not sure if it works the same way in C++. Maybe? I guess I could try and see if the build breaks 😄

Reply:
I also don't have a strong opinion about it.

> Also not sure if it works the same way in C++

It's apparently "translation phase 6" https://en.cppreference.com/w/cpp/language/translation_phases

Reply:
Doing this would make grepping the source for the string "System.Private.CoreLib.resources" (or "mscorlib.resources") a little harder.

String literal concatenation works the same way in both C and C++. It's used here: [[1]](https://github.com/mono/mono/blob/98320ecbde70b29db4c2591f9a8ae82ee60a2b52/mono/mini/llvm-jit.cpp#L460) [[2]](https://github.com/mono/mono/blob/98320ecbde70b29db4c2591f9a8ae82ee60a2b52/mono/mini/llvm-jit.cpp#L165)